packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -435,6 +435,20 @@ class _MenuAnchorState extends State<MenuAnchor> {
       );
     }
 
+    // If this menu anchor is not a menu bar, wrap the content into a Column
+    // to allow the child to use its preferred size.
+    // This is required when the menu anchor receives tight constraints.
+    // Otherwise the menu will be positioned below those constraints height
+    // and might appear detached from the menu anchor content when the content
+    // has an intrinsic height which is lower than the tight constraints height.
+    if (_orientation == Axis.vertical) {
+      child = Column(
```

Comment:
A column with a single child seems like an anti pattern? Can we use an Align instead?

Reply:
@navaronbracke I forgot to say thanks for your review :pray: 
I was not happy with this solution but tried so many things that having a solution that passes all tests was a relief :sweat_smile: 
A feedback stating that this is not a proper solution is very welcomed here, let me try to restate the problem and find other ways.